ddtrace/internal/core/__init__.py

```diff
@@ -168,15 +165,17 @@ class ExecutionContext:
     __slots__ = ["identifier", "_data", "_parents", "_span", "_token"]
 
     def __init__(self, identifier, parent=None, span=None, **kwargs):
-        _deprecate_span_kwarg(span)
+        if span is not None:
```

Comment:
This is redundant with `_deprecate_span_kwarg`'s internal logic

Reply:
Except at start where global context is not defined yet. If you remove that line, it will break at start.

Reply:
Found a fix by reversing logic order in the deprecate function